components/arm/collectors_test.go

```diff
@@ -0,0 +1,109 @@
+package arm
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	clk "github.com/benbjohnson/clock"
+	"github.com/edaniels/golog"
+	"github.com/golang/geo/r3"
+	v1 "go.viam.com/api/common/v1"
+	pb "go.viam.com/api/component/arm/v1"
+	"go.viam.com/test"
+
+	"go.viam.com/rdk/data"
+	"go.viam.com/rdk/resource"
+	"go.viam.com/rdk/spatialmath"
+	tu "go.viam.com/rdk/testutils"
+)
+
+const (
+	componentName   = "arm"
+	captureInterval = time.Second
+)
+
+var floatList = []float64{1.0, 2.0, 3.0}
+
+func TestCollectors(t *testing.T) {
+	tests := []struct {
+		name      string
+		params    data.CollectorParams
+		collector data.CollectorConstructor
+		expected  map[string]any
+	}{
+		{
+			name: "End position collector should write a pose",
+			params: data.CollectorParams{
+				ComponentName: componentName,
+				Interval:      captureInterval,
+				Logger:        golog.NewTestLogger(t),
+			},
```

Comment:
Minor but since the params are the same between the two tests, you can remove this field from the test struct and just construct them in the test loop below